File: src/main/java/com/wikia/tibia/repositories/ItemRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.ItemGateway;
import com.wikia.tibia.objects.Item;

public class ItemRepository extends WikiObjectRepository {

    public ItemRepository() {
        super(Item.class, new ItemGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/ItemRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.TibiaObject

class ItemRepository : WikiObjectRepository<TibiaObject>(
  wikiObjectClass = TibiaObject::class.java,
  wikiObjectGateway = WikiObjectGateway(Contract.ITEMS)
)
